rust/agama-server/src/network/model.rs

```diff
@@ -459,7 +514,85 @@ impl Default for Connection {
     }
 }
 
-#[derive(Default, Debug, PartialEq, Clone)]
+impl TryFrom<NetworkConnection> for Connection {
+    type Error = NetworkStateError;
+
+    fn try_from(conn: NetworkConnection) -> Result<Self, Self::Error> {
+        let id = conn.clone().id;
+        let mut connection = Connection::new(id, conn.device_type());
+
+        if let Some(method) = conn.clone().method4 {
+            let method: Ipv4Method = method.parse().unwrap();
```

Comment:
I am not sure whether this is the right place to do the parsing. What about doing that in the original `settings::NetworkConnection`? And, of course, `unwrapping` is not an option.